usaspending_api/etl/management/commands/combine_transaction_search_chunks.py

```diff
@@ -0,0 +1,218 @@
+
+        return create_temp_indexes, rename_indexes_old + rename_indexes_temp
+
+    def recreate_matview(self):
+        with connection.cursor() as cursor:
+            sql = (self.matview_dir / "componentized" / "transaction_search__create.sql").read_text()
+            cursor.execute(sql)
+
+    def insert_matview_data(self, chunk_count):
+        loop = asyncio.new_event_loop()
+        tasks = []
+
+        insert_table_sql = (self.matview_dir / "componentized" / "transaction_search__inserts.sql").read_text().strip()
+
+        for i, sql in enumerate(sqlparse.split(insert_table_sql)):
+            tasks.append(
+                asyncio.ensure_future(async_run_creates(sql, wrapper=Timer(f"Insert into table {i}"),), loop=loop,)
+            )
+
+        loop.run_until_complete(asyncio.gather(*tasks))
+        loop.close()
+
+    def create_constraints(self, constraint_definitions):
+
+        constraint_sql = "\n".join(constraint_definitions)
+
+        logger.debug(constraint_sql)
+
+        with connection.cursor() as cursor:
+            cursor.execute(constraint_sql)
+
+    def create_indexes(self, index_definitions, index_concurrency):
+        loop = asyncio.new_event_loop()
+        loop.run_until_complete(self.index_with_concurrency(index_definitions, index_concurrency))
+        loop.close()
+
+    async def index_with_concurrency(self, index_definitions, index_concurrency):
+        semaphore = asyncio.Semaphore(index_concurrency)
+        tasks = []
+
+        async def create_with_sem(sql, index):
+            async with semaphore:
+                return await async_run_creates(sql, wrapper=Timer(f"Creating Index {index}"),)
+
+        for i, sql in enumerate(index_definitions):
+            logger.info(f"Creating future for index: {i} - SQL: {sql}")
+            tasks.append(create_with_sem(sql, i))
+
+        return await asyncio.gather(*tasks)
+
+    @transaction.atomic
+    def swap_matviews(self, rename_indexes, rename_constraints):
+
+        swap_sql = (self.matview_dir / "componentized" / "transaction_search__renames.sql").read_text()
+
+        swap_sql += "\n".join(rename_indexes + rename_constraints)
+
+        logger.debug(swap_sql)
+
+        with connection.cursor() as cursor:
+            cursor.execute(swap_sql)
+
+    def remove_old_data(self, chunk_count):
+
+        drop_sql = (self.matview_dir / "componentized" / "transaction_search__drops.sql").read_text()
+
+        with connection.cursor() as cursor:
+            cursor.execute(drop_sql)
+
+    def empty_matviews(self):
+        empty_sql = (self.matview_dir / "componentized" / "transaction_search__empty.sql").read_text()
+
+        with connection.cursor() as cursor:
+            cursor.execute(empty_sql)
+
+    def grant_matview_permissions(self):
+        mods_sql = (self.matview_dir / "componentized" / "transaction_search__mods.sql").read_text()
+
+        with connection.cursor() as cursor:
+            cursor.execute(mods_sql)
```

Comment:
Not a blocker for PR merge.

All of these tiny functions could be refactored to contain less duplicate code. Additionally, it might be worthwhile to use existing functions like `execute_sql()` in usaspending_api/common/helpers/sql_helpers.py.